crates/primitives/src/transaction/mod.rs

```diff
@@ -174,14 +174,14 @@ impl Transaction {
 
     /// Gets the transaction's [`TxKind`], which is the address of the recipient or
     /// [`TxKind::Create`] if the transaction is a contract creation.
-    pub fn kind(&self) -> &TxKind {
+    pub fn kind(&self) -> TxKind {
```

Comment:
Not happy about not returning a reference here, but otherwise we would be returning a value referencing a temporary value in the match destructuring, any suggestion to fix this is very welcome!

Reply:
I think this is okay, because copy type,